docs/couler-step-zoo.md

```diff
@@ -0,0 +1,85 @@
+# Couler Step Zoo
+
+This document introduces the Couler Step Zoo, which consists of a collection of pre-defined and reusable steps that can
+be used directly as part of a workflow defined using Couler.
+
+
+## Existing Steps
+
+Currently, Couler Step Zoo consists of the following pre-defined steps:
+
+| Name | Description | API |
+| ---- | ----------- | --- |
+| MPI | Submit MPI-based distributed jobs via Kubeflow MPI Operator | `couler.steps.mpi.train()` |
+| TensorFlow  | Submit TensorFlow distributed training jobs via Kubeflow TF Operator | `couler.steps.tensorflow.train()` |
+| PyTorch  | Submit PyTorch distributed training jobs via Kubeflow PyTorch Operator | `couler.steps.pytorch.train()` |
+| Katib  | Submit AutoML experiments (e.g. hyperparameter tuning and neural architecture search) via Kubeflow Katib | `couler.steps.katib.run()` |
+
+If you'd like to add a new pre-defined steps to the Couler Step Zoo, please see the following sections for
+the specific requirements and instructions.
+
+
+## Requirements on New Steps
+
+In order to provide a consistent and friendly experience to our users, below are the requirements for a new pre-defined
+step to be eligible for inclusion in Couler Step Zoo:
+
+* It should be completely implemented using the core [Couler APIs](couler-api-design.md).
+* It should expose backend specific configurations instead of hard-coding them.
+* It should have a clear set of dependencies that can be easily installed with sufficient instructions.
```

Comment:
How to specify the dependencies for a new step function?  Adding the dependencies in [requirements.txt](https://github.com/couler-proj/couler/blob/master/requirements.txt) or other ways?